homework_unit11/main.js

```diff
@@ -0,0 +1,119 @@
+// Bai1: Viết 1 hàm trả về 1 array, nhận vào 2 tham số, tham số thứ 1 là ký tự cần lặp, tham số thứ 2 là số lần lặp:
+const createArr = (string, number) => {
+  const arr = []
+  for (let i = 0; i <= number; i++) {
+    arr.push(string)
+  }
+  return arr
+
+}
+
+//C2:
+
+const creArr = (string, number) => {
+  const arr = []
+  for (let i = 0; i < number; i++) {
+    arr.unshift(string)
+  }
+  return arr
+
+}
+
+// Bai2: Viết 1 hàm để đảo ngược 1 mảng, không được dùng hàm array.reverse():
+const reverseArr = array => {
+  const arr = []
+  for (let i = 0; i < array.length; i++) {
+    arr.unshift(array[i])
+  }
+  return arr
+
+}
+
+
+// Bai3: Xóa đi các giá trị được xem là sai:
+const deleteItem = array => {
+  const arr = []
+  for (let i = 0; i < array.length; i++) {
+    if ( array[i] ) {
+      arr.push(array[i])
+    }
+  }
+  return arr
+
+}
+
+//C2: Sử dụng filter:
+const filterItem = arr => arr.filter(Boolean)
+
+
+// Bai4: Tạo ra 1 array gồm các object có key và value tương ứng cặp array ban đầu:
+const convertArray = arr => {
+  const obj = {}
+  arr.forEach( array => {
+    obj[array[0]] = array[1]
+  });
+  return obj
+
+}
+
+
+// Bai5: Sắp xếp mảng tăng dần:
+const sortArr = arr => arr.sort( (a, b) => a - b )
+
+
+// Bai6: Kiểm tra input đầu vào có phải là object hay không?
+const testObject = obj => {
+  if (typeof(obj) === 'object' && !Array.isArray(obj)) {
+    return true
+  } else {
+    return false
+  }
+
```

Comment:
Không viết các lệnh dạng
```js
if (true) {
  return true
} else {
  return false
}
```

Sửa thành
```suggestion
  return typeof(obj) === 'object' && !Array.isArray(obj)
```